dashboard/app.R

```diff
@@ -224,40 +225,95 @@ server <- function(input, output, session) {
       return(NULL)
     }
   )
+  
+  return(s3bucket)
+}
 
-  # Get and prepare data
-  getData <- function(filename){
-    if(!is.null(s3bucket)) {
-      tryCatch(
-        {
-          s3readRDS(object = filename, bucket = s3bucket)
-        },
-        error = function(e) {
-          e
-          getFallbackData(filename)
-        }
-      )
-    } else {
-      getFallbackData(filename)
-    }
+getData <- function(filename, s3bucket){
+  if(!is.null(s3bucket)) {
+    tryCatch(
+      {
+        s3readRDS(object = filename, bucket = s3bucket)
+      },
+      error = function(e) {
+        e
+        getFallbackData(filename)
+      }
+    )
+  } else {
+    getFallbackData(filename)
   }
+}
 
-  getFallbackData = function(filename) {
-    path = ifelse(
-      file.exists(filename),
-      filename,
-      file.path("../dist/",filename)
-    )
-    readRDS(path)
+getFallbackData = function(filename) {
+  path = ifelse(
+    file.exists(filename),
+    filename,
+    file.path("../dist/",filename)
+  )
+  readRDS(path)
+}
+
+getAllData = function(s3bucket) {
+  dfStateCases <- getData("score_cards_state_cases.rds", s3bucket)
+  dfStateDeaths <- getData("score_cards_state_deaths.rds", s3bucket)
+  dfNationCases = getData("score_cards_nation_cases.rds", s3bucket)
+  dfNationDeaths = getData("score_cards_nation_deaths.rds", s3bucket)
+  dfStateHospitalizations = getData("score_cards_state_hospitalizations.rds", s3bucket)
+  dfNationHospitalizations = getData("score_cards_nation_hospitalizations.rds", s3bucket)
+  
+  # Pick out expected columns only
+  covCols = paste0("cov_", COVERAGE_INTERVALS)
+  expectedCols = c("ahead", "geo_value", "forecaster", "forecast_date",
+                   "data_source", "signal", "target_end_date", "incidence_period",
+                   "actual", "wis", "sharpness", "ae", "value_50",
+                   covCols)
+  
+  dfStateCases = dfStateCases %>% select(all_of(expectedCols))
+  dfStateDeaths = dfStateDeaths %>% select(all_of(expectedCols))
+  dfNationCases = dfNationCases %>% select(all_of(expectedCols))
+  dfNationDeaths = dfNationDeaths %>% select(all_of(expectedCols))
+  dfStateHospitalizations = dfStateHospitalizations %>% select(all_of(expectedCols))
+  dfNationHospitalizations = dfNationHospitalizations %>% select(all_of(expectedCols))
+  
+  df = rbind(dfStateCases, dfStateDeaths, dfNationCases, dfNationDeaths, dfStateHospitalizations, dfNationHospitalizations)
+  df = df %>% rename("10" = cov_10, "20" = cov_20, "30" = cov_30, "40" = cov_40, "50" = cov_50, "60" = cov_60, "70" = cov_70, "80" = cov_80, "90" = cov_90, "95" = cov_95, "98" = cov_98)
+  
+  return(df)
+}
+
+getRecentDataHelper = function() {
+  s3bucket <- getS3Bucket()
+  df <- data.frame()
+  
+  getRecentData = function() {
+    newS3bucket <- getS3Bucket()
+    
+    s3Contents <- s3bucket[attr(s3bucket, "names", exact=TRUE)]
```

Comment:
why do you need to fetch the content of the "old" s3 bucket each time? cannot you cache it similar to the caching of the data frame data itself?

Reply:
We could cache it as well, but it's relatively small and fast to compute so it doesn't *need* to be cached. I don't have a strong opinion about this and I'm not sure what's idiomatic in this situation.